### ISSUE TITLE
Context menu for elements inside Tabulator component

### ISSUE THREAD

**Issue by robert-cronin** [rjaros/kvision#54]
is it currently possible to add a context menu to the input of form elements? lets say a `TextInput`?
If so, how would I go about it? I'm not able to get it working using the way shown in the guide

**Comment by rjaros** [comment]
It works for me. There is a slight problem with closing the menu (it doesn't close automatically after click), but I've managed to make it work with a simple event handler. This is what I've added to the `FormTab.kt` in the `Showcase` example.
```
                Text(label = tr("Required text field with regexp [0-9] validator")).apply {
                    placeholder = tr("Enter your age")
                   // context menu
                    val contextMenu = ContextMenu {
                        header(tr("Menu header"))
                        link(tr("HTML"), "#!/basic")
                        link(tr("Forms"), "#!/forms")
                        setEventListener {
                            click = {
                                self.hide()
                            }
                        }
                    }
                    setContextMenu(contextMenu)
                   // context menu
                },
```
It works fine with `TextInput` component as well.

**Comment by robert-cronin** [comment]
Perhaps it would be useful to tell you that I am trying to use it in a custom `Tabulator`s `formatterComponentFunction` and a custom component.

I pasted in that exact code and it still didn't work, perhaps its something to do with the `Tabulator`? since I also can't get a context menu to work with `SimplePanel`

To replicate my issue, you could use this code:
```

@Serializable
data class SomeAgeClass(var age: Int = 0)
        
var someAgeClassList:ObservableList<SomeAgeClass> = observableListOf(
        SomeAgeClass(18),
        SomeAgeClass(19),
        SomeAgeClass(17)
)

tabulator(
        someAgeClassList,
        options = TabulatorOptions(
                reactiveData = true,
                layout = Layout.FITCOLUMNS,
                columns = listOf(
                        ColumnDefinition("Age", "", headerSort = false,
                                formatterComponentFunction = { _, _, data: SomeAgeClass ->
                                    Text(label = tr("Required text field with regexp [0-9] validator"), value = data.age.toString()).apply {
                                        placeholder = tr("Enter your age")
                                        // context menu
                                        val contextMenu = ContextMenu {
                                            header(tr("Menu header"))
                                            link(tr("HTML"), "#!/basic")
                                            link(tr("Forms"), "#!/forms")
                                            setEventListener {
                                                click = {
                                                    self.hide()
                                                }
                                            }
                                        }
                                        setContextMenu(contextMenu)
                                        // context menu
                                    }
                                }
                        )
                )
        )
)
```

**Comment by rjaros** [comment]
Tabulator overwrites a lot of mouse event handlers for its DOM elements. And it's probably just overwrites right mouse click as well. I've been struggling with this event handlers trying to make `Select` control work inside Tabulator, but unfortunately I didn't manage to do it. It could be similar issue. But I will take a closer look at this.

**Comment by robert-cronin** [comment]
I thought it might be a tabulator thing, the context menu is working fine for me everywhere else.
I've just discovered cell click callbacks in this webpage: http://tabulator.info/docs/4.2/callbacks
this would do it for me, except other users might need to actually overwrite this and have the context menu directly on the custom Component

**Comment by robert-cronin** [comment]
I'm not sure but maybe this will help as well: https://github.com/olifolkerd/tabulator/issues/1242

**Comment by rjaros** [comment]
I've made some tests and it turns out not to be the case of events. The "contextmenu" event is triggered correctly and the context menu is in fact showing. But it cannot be seen, because it's showing inside incorrect component tree and in incorrect position. I've commited some fixes and after 0.0.37 version is released, the following code can be used to open a context menu inside a tabulator component (note that the `ContextMenu` component must be created inside the `onRendered` callback and it will be shown in a fixed position):
```
                    ColumnDefinition(
                        "Name", "name",
                        editorComponentFunction = { _, onRendered, success, _, data ->
                            val textInput = TextInput(value = data.name).apply {
                                size = InputSize.SMALL
                                setEventListener<TextInput> {
                                    change = {
                                        success(self.value)
                                    }
                                }
                            }
                            onRendered {
                                val contextMenu = ContextMenu(textInput) {
                                    header(tr("Menu header"))
                                    link(tr("HTML"), "#!/basic")
                                    link(tr("Forms"), "#!/forms")
                                    setEventListener {
                                        click = {
                                            self.hide()
                                        }
                                    }
                                }
                                textInput.setContextMenu(contextMenu)
                            }
                            textInput
                        }
                    )
```

**Comment by robert-cronin** [comment]
great news! thanks Robert
